### PR TITLE
unittests/tests-rpl_srh: various improvements

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
During my [refactoring work of IPv6 reception](https://github.com/RIOT-OS/RIOT/projects/24) I also improved upon the unittests for the RPL SRH handler, but I didn't find any other way to include them (also their usefulness, except for compile tests [but there also `tests/gnrc_ipv6_ext` can be used] are little), so here they are in a separate PR.

* Some stack optimizations
* Some refactoring
* Adding some tests for error cases (!)
* Rename to `tests-gnrc_rpl_srh`

### Testing procedure
Run `tests/unittests` either as a whole, or just `tests-gnrc_rpl_srh`.

### Issues/PRs references
Related to my refactoring work (see [project](https://github.com/RIOT-OS/RIOT/projects/24), but not a dependency of it nor dependent on any PRs in it (but helpful for testing #10227 ;-)).

### CODE REVIEW

**Comment by miri64** [comment]
Ping @cgundogan?

**Comment by jia200x** [comment]
I'm not an expert in RPL :(
@cgundogan could you help with the `Reviewed: 1-Fundamentals` label while I do the rest? (e.g tests make sense, etc)

**Comment by miri64** [comment]
You don't have to be an expert on RPL to read the API of that function and see if the tests test all cases ;-).

File: tests/unittests/tests-gnrc_rpl_srh/tests-gnrc_rpl_srh.c [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    srh->len = (2 * sizeof(ipv6_addr_t)) / 8;
+    srh->seg_left = SRH_SEG_LEFT;
+    memcpy(vec, &a1, sizeof(a1));
+    memcpy(vec + sizeof(a1), &a2, sizeof(a2));
+
+    res = gnrc_rpl_srh_process(&hdr, srh);
+    TEST_ASSERT_EQUAL_INT(res, GNRC_IPV6_EXT_RH_ERROR);
+}
+
+static void test_rpl_srh_route_multicast(void)
```
Comment (jia200x) [inline tests/unittests/tests-gnrc_rpl_srh/tests-gnrc_rpl_srh.c]: shouldn't these functions have `test_gnrc` prefix?

Comment (miri64) [reply tests/unittests/tests-gnrc_rpl_srh/tests-gnrc_rpl_srh.c]: It's static, so the prefix is arbitrary (it's called by the unit test runner via the [array defined below](https://github.com/RIOT-OS/RIOT/blob/a63dec807435fae83de2447663d307571aeb32e2/tests/unittests/tests-gnrc_rpl_srh/tests-gnrc_rpl_srh.c#L185-L191)). Other test suites use a similar naming scheme (see e.g. `tests-gnrc_ipv6_nib`)

Comment (jia200x) [reply tests/unittests/tests-gnrc_rpl_srh/tests-gnrc_rpl_srh.c]: ok :)

**Comment by miri64** [comment]
Oh, this information might also be helpful to understand the unittests :-) https://tools.ietf.org/html/rfc6554#section-3

File: tests/unittests/tests-gnrc_rpl_srh/tests-gnrc_rpl_srh.c [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+static inline void _init_hdrs(gnrc_rpl_srh_t **srh, uint8_t **vec,
+                              const ipv6_addr_t *dst)
+{
+    *srh = (gnrc_rpl_srh_t *)buf;
+    *vec = (uint8_t *)(*srh + 1);
+    memcpy(&hdr.dst, dst, sizeof(hdr.dst));
+}
+
+static void test_rpl_srh_dst_multicast(void)
```
Comment (jia200x) [inline tests/unittests/tests-gnrc_rpl_srh/tests-gnrc_rpl_srh.c]: This and test below make sense:

```
@RFC6554

   Multicast addresses MUST NOT appear in an SRH or in the IPv6
   Destination Address field of a datagram carrying an SRH.
```

Comment (jia200x) [reply tests/unittests/tests-gnrc_rpl_srh/tests-gnrc_rpl_srh.c]: (notes to self)

File: tests/unittests/tests-rpl_srh/tests-rpl_srh.c [hunk 3]
```diff
@@ -61,6 +68,60 @@ static inline void _init_hdrs(gnrc_rpl_srh_t **srh, uint8_t **vec,
...
+    srh->len = (2 * sizeof(ipv6_addr_t)) / 8;
+    srh->seg_left = SRH_SEG_LEFT;
+    memcpy(vec, &mcast, sizeof(mcast));
+    memcpy(vec + sizeof(mcast), &a1, sizeof(a1));
+
+    res = gnrc_rpl_srh_process(&hdr, srh);
+    TEST_ASSERT_EQUAL_INT(res, GNRC_IPV6_EXT_RH_ERROR);
+}
+
+static void test_rpl_srh_too_many_seg_left(void)
```
Comment (jia200x) [inline tests/unittests/tests-rpl_srh/tests-rpl_srh.c]: ok, all new tests make sense

**Comment by jia200x** [comment]
`vera++` doesn't report errors in this PR. Code style OK

**Comment by jia200x** [comment]
tested on `native` and `b-l072z-lrwan1`. Works as expected

**Comment by miri64** [comment]
Thanks for the review! :-)

**Pull request merged**